### PR TITLE
feat(multibit): implement deterministic multibit pbs

### PR DESCRIPTION
**Pull request by IceTDrinker**

### CODE REVIEW

File: tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs [hunk 1]
```diff
@@ -432,3 +817,87 @@ pub fn test_lwe_encrypt_std_multi_bit_pbs_decrypt_factor_3_thread_12_custom_mod(
...
+            glwe_dimension: GlweDimension(2),
+            polynomial_size: PolynomialSize(1024),
+            glwe_modular_std_dev: StandardDev(0.0000000000000003152931493498455),
+            message_modulus_log: CiphertextModulusLog(3),
+            ciphertext_modulus: CiphertextModulus::try_new_power_of_2(63).unwrap(),
+            grouping_factor: LweBskGroupingFactor(3),
+            thread_count: ThreadCount(12),
+        },
+    );
+}
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:903]: could factorize multi bit params

**Comment by IceTDrinker** [comment]
might revert the renaming commit as in low level layers the PBS takes a LUT as input which then is copied and serves as the accumulator for the blind rotation

**Comment by IceTDrinker** [comment]
will add shortint deterministic choice for the server key

**Review by tmontaigu — APPROVED** [review]

File: tfhe/src/core_crypto/algorithms/lwe_multi_bit_programmable_bootstrapping.rs [hunk 2]
```diff
@@ -560,6 +557,310 @@ pub fn multi_bit_blind_rotate_assign<Scalar, InputCont, OutputCont, KeyCont>(
...
+        .zip(ggsw_vec.chunks_exact(ggsw_per_multi_bit_element.0))
+    {
+        work_queue.push((lwe_mask_elements, ggsw_group));
+    }
+
+    assert!(work_queue.len() == lwe_mask.lwe_dimension().0 / grouping_factor.0);
+
+    let work_queue = &work_queue;
+
+    // At any point in time the idea is to have half the buffer already processed ready for the
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/core_crypto/algorithms/lwe_multi_bit_programmable_bootstrapping.rs:634]: I'll do it, just for you :D

**Comment by IceTDrinker** [comment]
@slab-ci cpu_multi_bit_test

**Comment by IceTDrinker** [comment]
follow-up issue to plug in shortint https://github.com/zama-ai/tfhe-rs-internal/issues/168 to avoid having this PR rotting

**Pull request merged**